src/Native/System.Native/pal_networking.cpp

```diff
@@ -379,7 +380,119 @@ static void ConvertHostEntPlatformToPal(HostEntry& hostEntry, hostent& entry)
     }
 }
 
-#if HAVE_GETHOSTBYNAME_R
+#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR && (!HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R)
```

Comment:
It looks like right now there are two sections:
```C++
#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR && (!HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R)
...
#endif
#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR && (!HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R)
...
#endif
```
I wonder if it'd be cleaner if this were split up, e.g.
```
#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR
#if !HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R
...
#else
...
#endif // !HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R
#endif // !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR
```
?

Also, what about the case where HAVE_GETHOSTBYNAME_R but !HAVE_GETHOSTBYADDR_R, and vice versa?  Or does that not happen?  I may be misreading, but it looks like we assume that if one exists the other also exists... is that a fair assumption?